frontend/src/v5/ui/controls/shareTextField/shareTextField.component.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ *  Copyright (C) 2021 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+
+import { InputAdornment } from '@material-ui/core';
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+import { formatMessage } from '@/v5/services/intl';
+import {
+	LinkLabel,
+	LinkBar,
+	CopyToClipboardIconContainer,
+	CopyToClipboardIcon,
+	Tick,
+	CopyToClipboardTooltip,
+	CopiedToClipboardTooltip,
+} from './shareTextField.styles';
+
+type IShareTextField = {
+	label: string,
+	text: string,
+};
+
+const IS_COPYING_DURATION_MS = 3000;
+
+export const ShareTextField = ({ label, text }: IShareTextField) => {
```

Comment:
can we rename the prop text to value? Just to keep the same interface as `<Textfield />`